test/fail_compilation/test17275.d

```diff
@@ -0,0 +1,19 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/test17275.d(12): Error: undefined identifier 'ModuleGroup', did you mean function 'moduleGroup'?
+fail_compilation/test17275.d(12): Error: inout on return means inout must be on a parameter as well for inout(ModuleGroup)()
```

Comment:
There's already test/fail_compilation/fail17275.d (https://github.com/dlang/dmd/pull/6690/files#diff-278b7365e40009be458ef32b7f8d85a3). Why don't you put the test output in there?